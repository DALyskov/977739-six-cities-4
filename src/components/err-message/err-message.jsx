import React from 'react';
import propTypes from 'prop-types';
import {connect} from 'react-redux';
import {getErrMessage} from '../../reducer/data/selectors.js';

const ErrMessage = (props) => {
  const {errMessage} = props;
  return (
    <div
      className="err-message"
      style={{
        margin: `10px auto`,
        padding: `0 60px`,
        color: `red`,
        textAlign: `center`,
      }}>
      {errMessage}
    </div>
  );
};

ErrMessage.propTypes = {
  errMessage: propTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  errMessage: getErrMessage(state),
});

export {ErrMessage};
export default connect(mapStateToProps)(ErrMessage);
