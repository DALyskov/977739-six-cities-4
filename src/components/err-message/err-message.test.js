import React from 'react';
import renderer from 'react-test-renderer';

import {ErrMessage} from './err-message.jsx';

describe(`ErrMessage_snapchots`, () => {
  it(`ErrMessage_with_data`, () => {
    const tree = renderer.create(<ErrMessage errMessage={`abcd`} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
