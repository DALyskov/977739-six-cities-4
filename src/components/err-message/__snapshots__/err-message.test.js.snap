// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrMessage_snapchots ErrMessage_with_data 1`] = `
<div
  className="err-message"
  style={
    Object {
      "color": "red",
      "margin": "10px auto",
      "padding": "0 60px",
      "textAlign": "center",
    }
  }
>
  abcd
</div>
`;
